src/uri.rs

```diff
@@ -64,21 +64,29 @@ impl<T: DeserializeOwned> Uri<T> {
     /// ```
     pub async fn fetch(&self) -> crate::Result<T> {
         match self.fetch_raw().await {
-            Ok(response) => match response.status().as_u16() {
-                200..=299 => Ok(response.json().await?),
-                status => Err(Error::HttpError(StatusCode::from(status))),
+            Ok(mut response) => {
+                let status = response.status();
+                if status.is_success() {
+                    Ok(response.body_json().await?)
+                } else {
+                    Err(Error::HttpError(StatusCode::from(u16::from(status))))
```

Comment:
Does surf use `http` crate status code? If yes I think we should drop the httpstatus dependency as well

Reply:
Surf uses [http_types](https://docs.rs/http-types/latest/http_types/).
This would also create additional backwards incompatibility since it leaks out too in case that matters. Same group that created surf.

Reply:
Hmm, I think we should take the opportunity of making a breaking change and think about other good changes. My understanding is that `http` is the most standard source of basic http types. `httpstatus` is definitely not it. Do you think http_types is a good idea to be the type we export out of this crate's error type?

Reply:
On the topic of making breaking changes, scryfall itself makes breaking changes all the time, so this crate has to make a lot of them as well. To be precise, wizards makes a lot of breaking changes 🤣

Reply:
I can't say I have a ton of experience with it. I think it is better quality than `httpstatus` but haven't done a thorough comparison either. From a quick survey, `StatusCode` in `http-types` and `http` looks pretty similar.

Options:
* `http-types` - zero impedance mismatch with `surf`
* `http` - most popular interface
* `u16` - Remove the public facing dep entirely as just return a `u16`, let the client decide if they want to "interpret" it by pulling it into their favorite `StatusCode` object. Prevents consumers from being pinned to a specific version of the library too.

I actually think the last option might be the best.

Reply:
I think we should go with the last option them. Makes sense to me